studies/models.py

```diff
@@ -776,6 +779,20 @@ def generate_videos_from_events(self):
         return Video.objects.bulk_create(video_objects)
 
 
+@receiver(post_save, sender=Response)
+def take_action_on_exp_data(sender, instance, created, **kwargs):
+    """Performs post-save actions based on the current frame.
```

Comment:
Just a warning about relying on this hook for marking consent footage - if saving the response after a single frame fails for whatever reason, the participant is allowed to continue and the frameplayer will just try again next frame (which will mean adding multiple frames at once). So we might want to consider the edge case where save fails immediately after the consent frame, but then succeeds later, so that it's never the most recent frame. This is less relevant for EXIT type frames, which should come at the end of the study.

Reply:
Fair point. It's enough to keep it as is.